neuralmonkey/config/utils.py

```diff
@@ -30,13 +33,13 @@ def dep_func(*args, **kwargs):
 initialize_vocabulary = vocabulary.initialize_vocabulary
 
 
-def adam_optimizer(learning_rate=1e-4):
+def adam_optimizer(learning_rate=1e-4) -> tf.train.AdamOptimizer:
     return tf.train.AdamOptimizer(learning_rate)
 
 
-def adadelta_optimizer(**kwargs):
+def adadelta_optimizer(**kwargs) -> tf.train.AdadeltaOptimizer:
     return tf.train.AdadeltaOptimizer(**kwargs)
 
 
-def variable(initial_value=0, trainable=False, **kwargs):
+def variable(initial_value=0, trainable=False, **kwargs) -> tf.Variable:
```

Comment:
Jenomže Union[int, Tensor] je málo, může to být taky float, numpy array, asi i list floatů, nevím jestli to může být nějaký uživatelem definovaný typ, ale asi by to šlo zařídit. Zkrátka potřeboval bych nějaké rozhraní, které mi říká, že je to konvertovatelné na Tensor. Nevím jestli něco takového někde je, ale připadá mi zbytečné to řešit. Mnohem důležitější, než mít všechno přesně anotované je to, aby to vůbec bylo anotované a kontrolovalo se to (teď se těla mnoha funkcí nekontrolují, protože mypy kontroluje jenom anotované funkce). A v případě, že nejsem schopný poskytnout přesný typ, tak je správně být na vstupu permisivní, abych někomu nezakazoval platný kód.